src/bloqade/analysis/fidelity/impls.py

```diff
@@ -0,0 +1,77 @@
+from kirin import interp
+from kirin.lattice import EmptyLattice
+from kirin.analysis import const
+from kirin.dialects import scf
+from kirin.dialects.scf.stmts import For, Yield, IfElse
+
+from .analysis import FidelityAnalysis
+
+
+@scf.dialect.register(key="circuit.fidelity")
+class ScfFidelityMethodTable(interp.MethodTable):
```

Comment:
For this PR since we will be breaking the interpreter for regions in the next version of kirin can we remove the SCF interpretation here?

Reply:
As in, remove the entire method table? Or what do you mean?

Reply:
Yeah we don't need to worry about control flow for this PR